File: 1_dfs_bfs/김민주/타겟넘버.cpp
#include <string>
#include <vector>

using namespace std;

// 전역 변수
int ans = 0;
int target;

void dfs(int i, int sum, vector<int> numbers);

int solution(vector<int> numbers, int t) {
    target = t;

    dfs(0, 0, numbers);

    return ans;
}

void dfs(int i, int sum, vector<int> numbers){
    if (i == numbers.size()){
        if (sum == target){
            ans += 1;
        }
        return;
    }
    dfs(i + 1, sum + numbers[i], numbers);
    dfs(i + 1, sum - numbers[i], numbers);
    
}

File: 1_dfs_bfs/강대훈/타겟넘버.swift
func solution(_ numbers: [Int], _ target: Int) -> Int {
    var ans = 0

    dfs(0, 0)

    func dfs(_ idx: Int, _ sum: Int) {
        if idx == numbers.count {
            if sum == target {
                ans += 1
            }
            return
        }

        dfs(idx + 1, sum + numbers[idx])
        dfs(idx + 1, sum - numbers[idx])
    }

    return ans
}
